src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -541,10 +543,30 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				event.preventDefault();
 				return false;
 			}
-
-			// Skip processing by xterm.js of keyboard events that resolve to commands described
-			// within commandsToSkipShell
-			if (resolveResult && this._skipTerminalCommands.some(k => k === resolveResult.commandId)) {
+			const promptChoices: IPromptChoice[] = [
```

Comment:
Move this inside the if block so we don't create these objects on every keypress unless needed